Sources/FluentKit/Query/Builder/QueryBuilder.swift

```diff
@@ -229,12 +234,29 @@ public final class QueryBuilder<Model>
                 }
             }
         }
-        
-        // If deleted models aren't included, add filters
-        // to exclude them for each model being queried.
-        if !self.includeDeleted {
-            for model in self.models {
-                model.excludeDeleted(from: &query)
+
+        for model in self.models {
+            let initializedModel = model.init()
+            
+            // If deleted models aren't included, add filters
+            // to exclude them for each model being queried.
+            if !self.includeDeleted {
+                model.excludeDeleted(initialized: initializedModel, from: &query)
+            }
+            
+            // If we don't want to force delete, just touch
+            // the timestamp model. We also change it to an
+            // update query. If it is a force delete, still
+            // touch the timestamps
+            if !self.forceDelete, let _ = initializedModel.deletedTimestamp {
+                initializedModel.touchTimestamps(.update, .delete)
+                query.action = .update
+                query.input = [.dictionary(initializedModel.input.values)]
```

Comment:
I'm not sure this makes sense inside of `self.models`. I don't think Fluent supports deleting with joins yet or if the underlying DBs even support that. I think you are safe to assume here that the only model we care about during `delete` is the generic `Model`.